docs/HowToGuides/GettingStarted.md

```diff
@@ -0,0 +1,603 @@
+  on the commandline for more fine-grained control over which exact tests are
+  run.
+<!-- TODO: Insert SourceKit/stdlib specific instructions? -->
+
+### Editing
+
+Make changes to the code as appropriate. Implement a shiny new feature!
+Or fix a nasty bug! Update the documentation as you go! <!-- please 🙏 -->
+The codebase is your oyster!
+
+:construction::construction_worker::building_construction:
+
+Now that you have made some changes, you will need to rebuild...
+
+### Incremental rebuilds with Ninja
+
+To rebuild the compiler:
+```sh
+ninja -C ../build/Ninja-RelWithDebInfoAssert/swift-macosx-x86_64 swift-frontend
+```
+
+To rebuild everything, including the standard library:
+```sh
+ninja -C ../build/Ninja-RelWithDebInfoAssert/swift-macosx-x86_64
+```
+
+### Incremental builds with Xcode
+
+Rebuilding works the same way as with any other Xcode project; you can use
+<kbd>⌘</kbd>+<kbd>B</kbd> or Product → Build.
+
+### Spot checking an incremental build
+
+As a quick test, go to `lib/Basic/Version.cpp` and tweak the version
+printing code slightly. Next, do an incremental build as above. This incremental
+build should be much faster than the from-scratch build at the beginning.
+Now check if the version string has been updated:
+
+```sh
+../build/Ninja-RelWithDebInfoAssert/swift-macosx-x86_64/bin/swift-frontend --version
+```
+
+This should print your updated version string.
+
+## Reproducing an issue
+
+Starter bugs typically have small code examples that fit within a single file.
+You can reproduce such an issue in various ways, such as compiling it from the
+commandline using `/path/to/swiftc MyFile.swift`, pasting the code into
+[Compiler Explorer][] (aka godbolt) or using an Xcode Playground.
+
+[Compiler Explorer]: https://godbolt.org
+
+For files using frameworks from an SDK bundled with Xcode, you need the pass
+the SDK explicitly. Here are a couple of examples:
+```sh
+# Compile a file to an executable for your local machine.
+xcrun -sdk macosx /path/to/swiftc MyFile.swift
+
+# Say you are trying to compile a file importing an iOS-only framework.
+xcrun -sdk iphoneos /path/to/swiftc -target arm64-apple-ios13.0 MyFile.swift
+```
+You can see the full list of `-sdk` options using `xcodebuild -showsdks`,
+and check some potential `-target` options for different operating systems by
+skimming the compiler's test suite under `test/`.
+
+Sometimes bug reports come with SwiftPM packages or Xcode projects as minimal
+reproducers. While we do not add packages or projects to the compiler's test
+suite, it is generally helpful to first reproduce the issue in context before
+trying to create a minimal self-contained test case. If that's the case with
+the bug you're working on, check out our
+[instructions on building packages and Xcode projects with a locally built compiler](/docs/HowToGuides/FAQ.md#how-do-i-use-a-locally-built-compiler-to-build-x).
+
+## Running tests
```

Comment:
This section should point out that you should compare the failures you see after changes to the baseline (brought up in the Building section), in case there are baseline failures. Otherwise, you might end up thinking you did something wrong, even though all the failures you are seeing are from the baseline.